packages/gatsby-source-filesystem/src/create-lazy-file-node.js

```diff
@@ -0,0 +1,143 @@
+"use strict"
+
+const path = require(`path`)
+
+const mime = require(`mime`)
+const {
+  getRemoteFileExtension,
+  getRemoteFileName,
+  createFilePath,
+} = require(`./utils`)
+
+const { createContentDigest, slash } = require(`gatsby-core-utils`)
+const CACHE_DIR = `.cache`
+const FS_PLUGIN_DIR = `gatsby-source-filesystem`
+
+function createFileNode(pathToFile, createNodeId, pluginOptions = {}) {
+  const slashed = slash(pathToFile)
+  const parsedSlashed = path.parse(slashed)
+  const slashedFile = Object.assign({}, parsedSlashed, {
+    absolutePath: slashed,
+    // Useful for limiting graphql query with certain parent directory
+    relativeDirectory: slash(
+      path.relative(pluginOptions.path || process.cwd(), parsedSlashed.dir)
+    ),
+  })
+  let internal
+
+  const mediaType = mime.getType(slashedFile.ext)
+  const contentDigest = createContentDigest(pathToFile)
+  internal = {
+    contentDigest,
+    type: `RemoteFile`,
+    mediaType: mediaType ? mediaType : `application/octet-stream`,
+    description: `File "${path.relative(process.cwd(), slashed)}"`,
+  }
+
+  return JSON.parse(
+    JSON.stringify(
+      Object.assign(
+        {
+          // Don't actually make the File id the absolute path as otherwise
+          // people will use the id for that and ids shouldn't be treated as
+          // useful information.
+          id: createNodeId(pathToFile),
+          children: [],
+          parent: null,
+          internal,
+          sourceInstanceName: pluginOptions.name || `__PROGRAMMATIC__`,
+          absolutePath: slashedFile.absolutePath,
```

Comment:
So I guess the spread of `slashedFile` goes firstand then you can drop this line :)